numba/np/random/random_methods.py

```diff
@@ -0,0 +1,349 @@
+import numpy as np
+
+from numba.core.extending import register_jitable
+
+from numba.np.random._constants import (UINT32_MAX, UINT64_MAX,
+                                        UINT16_MAX, UINT8_MAX)
+from numba.np.random.generator_core import next_uint32, next_uint64
+
+# All following implementations are direct translations from:
+# https://github.com/numpy/numpy/blob/7cfef93c77599bd387ecc6a15d186c5a46024dac/numpy/random/src/distributions/distributions.c
+
+
+@register_jitable
+def gen_mask(mask):
+    mask = mask | (mask >> 1)
+    mask = mask | (mask >> 2)
+    mask = mask | (mask >> 4)
+    mask = mask | (mask >> 8)
+    mask = mask | (mask >> 16)
+    mask = mask | (mask >> 32)
+    return mask
+
+
+@register_jitable
+def buffered_bounded_bool(bitgen, off, rng, mask, bcnt, buf):
```

Comment:
why is `mask` unused?

Reply:
Mask will be used in methods being added in #8042. Methods in both PR use same underlying functions in Numpy differently, so I added them as-is for now. I'll tag this comment in that PR just to be safe.